2015-12-02 review

mfulton26 opened the pull request:
- `for in line` can iterate over lines in a string
- string.split(delimiter) can be used to split a string
- [expr for i in list] can be used to transform a list
- sorted(list) can also be used to sort a list
- lists (tuples) can be destructured to variables

Commit:
2015-12-02 review

- `for in line` can iterate over lines in a string
- string.split(delimiter) can be used to split a string
- [lambda for i in list] can be used to transform a list
- sorted(list) can also be used to sort a list
- lists (tuples) can be destructured to variables

mfulton26 commented:
FYI: I found your solution, although longer than perhaps needed, very solid, it shows a very good grasp of some fundamentals in the way code works (every language has its "nice-to-haves" built on top of this knowledge and learning those vast tools available takes time)

nice work 👍

FormulaTranslator commented:
This is great, just the stuff I was looking for! I had this thought that I was brute forcing a lot of this because I didn't understand the language as well as I should. Also, my file/string handling skills are pretty minimal (as if it wasn't apparent). So these tips are very helpful. Thanks!!

mfulton26 commented on Project2_2015.py:
```diff
@@ -1,40 +1,14 @@
-            else:
-                current_number += char
+    for line in file:
+        l, w, h = sorted([int(i) for i in line.split('x')])
```
in retrospect these could be called anything and in any order (however you want to perceive a 3d object where largest dimensions come first); e.g. `width, height, depth` might be more natural and better naming than `w,  h, d` (or `l, w, h` as it is currently)

mfulton26 replied:
also obviously they don't need to be destructured but saying `dimensions[0]` all the time gets old (IMO) 😄

mfulton26 replied:
`d0, d1, d2`? 🤷

FormulaTranslator replied:
so I took one CS course in college and the professor was anal about names. He hated abbreviations, because others would have a hard time reading it. In your experience, what is typically used, full names or abbreviations?

mfulton26 replied:
full names except for let maybe like `i` for `index` in for loops (but really that type of code is generally avoided for things like `for line in file`, etc.

mfulton26 commented:
quick note on day 2 I didn't think of before, as an alternative to sorting one could use `min(a, b, c)`:

```patch
diff --git a/Project2_2015.py b/Project2_2015.py
index 6d56713..b4ac153 100644
--- a/Project2_2015.py
+++ b/Project2_2015.py
@@ -3,13 +3,14 @@ total_ribbon = 0
 
 with open('Project2_2015', 'r') as file:
     for lines in file:
-        length, width, height = sorted(int(i) for i in lines.split('x'))
+        length, width, height = [int(i) for i in lines.split('x')]
         area = 2 * (length * width + width * height + height * length)
-        smallest_side = length * width
+        smallest_side = min(length * width, width * height, length * height)
         total_wrapper += area + smallest_side
 
         box_volume = length * width * height
-        smallest_perimeter = 2 * (length + width)
+        smallest_perimeter = 2 * \
+            min(length + width, width + height, length + height)
         total_ribbon += box_volume + smallest_perimeter
 
 print(total_wrapper)
```
